index.js

```diff
@@ -17,7 +25,44 @@ function isEmpty(obj){
     return true
 }
 
-function createXHR(options, callback) {
+function normalizeArgs(arg1, arg2, arg3) {
+    var options, callback;
+
+    if (typeof arg1 === "string") {
+        if (typeof arg2 === "object" && typeof arg3 === "function") {
+            // xhr(url, options, cb)
+            options = xtend(arg2)
+            options.uri = arg1
+            callback = arg3
+        } else {
+            // xhr(url, cb)
+            options = { uri: arg1 }
+            callback = arg2
+        }
+    } else {
+      // xhr(options, cb)
```

Comment:
Yeah, I know. [here's request's param parsing](https://github.com/request/request/blob/master/index.js#L26-L42) for reference.

Request names all their arguments, but depending on the way it's called, `(uri, opts, cb)` or `(opts, cb)`, the names begin to not represent what they actually are, so I just went with arg1, arg2, arg3.

I know it's long, but I thought this was clearer than the alternatives. Let me know if you want something along the lines of how `request` does it.